lib/opal/source_map/file.rb

```diff
@@ -0,0 +1,188 @@
+  #
+  # 1. The zero-based starting column of the line in the generated code that
+  #    the segment represents. If this is the first field of the first segment, or
+  #    the first segment following a new generated line (“;”), then this field
+  #    holds the whole base 64 VLQ. Otherwise, this field contains a base 64 VLQ
+  #    that is relative to the previous occurrence of this field. Note that this
+  #    is different than the fields below because the previous value is reset
+  #    after every generated line.
+  #
+  # 2. If present, an zero-based index into the “sources” list. This field is
+  #    a base 64 VLQ relative to the previous occurrence of this field, unless
+  #    this is the first occurrence of this field, in which case the whole value
+  #    is represented.
+  #
+  # 3. If present, the zero-based starting line in the original source
+  #    represented. This field is a base 64 VLQ relative to the previous
+  #    occurrence of this field, unless this is the first occurrence of this
+  #    field, in which case the whole value is represented. Always present if
+  #    there is a source field.
+  #
+  # 4. If present, the zero-based starting column of the line in the source
+  #    represented. This field is a base 64 VLQ relative to the previous
+  #    occurrence of this field, unless this is the first occurrence of this
+  #    field, in which case the whole value is represented. Always present if
+  #    there is a source field.
+  #
+  # 5. If present, the zero-based index into the “names” list associated with
+  #    this segment. This field is a base 64 VLQ relative to the previous
+  #    occurrence of this field, unless this is the first occurrence of this
+  #    field, in which case the whole value is represented.
+  def segment_from_fragment(fragment, generated_column)
+    source_index     = 0                          # always 0, we're dealing with a single file
+    original_line    = fragment.line - 1          # fragments have 1-based lines
+    original_column  = fragment.column            # fragments have 0-based columns
+    if fragment.source_map_name
+      map_name_index   = (@names_map[fragment.source_map_name.to_s] ||= @names_map.size)
+
+      [
+        generated_column,
+        source_index,
+        original_line,
+        original_column,
+        map_name_index,
+      ]
+    else
+      [
+        generated_column,
+        source_index,
+        original_line,
+        original_column,
+      ]
+    end
+  end
+
+  def relative_mappings
+    @relative_mappings ||= begin
+      reference_segment = [0, 0, 0, 0, 0]
+      absolute_mappings.map do |absolute_mapping|
+        # [generated_column, source_index, original_line, original_column, map_name_index]
+        reference_segment[0] = 0 # reset the generated_column at each new line
+
+        absolute_mapping.map do |absolute_segment|
+          segment = []
+
+          segment[0] = absolute_segment[0].to_int -  reference_segment[0].to_int
```

Comment:
It's an ensurance agains non-integers, I get the TypeError for free, but I guess that's kinda debugging code, although it doesn't harm to keep it, wdyt?

Reply:
I don't get it. `to_int` is defined only on `Integer` and `Float`. For `String` class MRI throws a `NoMethodError`. Which object is in the `absolute_segment[0]`?

Reply:
I was wrong about the TypeError 😓 
Anyway it was a guard against `nil`s most probably, I use it from time to time when I expect an integer and I want to avoid the automatic conversions I'd get from `nil`/`String`/etc.
Think of it as an assertion I expect an integer there.

I don't have strong feelings, if you find it's confusing it's probably better to remove it